storjscandb/dbx/storjscandb.dbx.go

```diff
@@ -323,6 +323,12 @@ func (obj *pgxDB) Schema() string {
 	created_at timestamp with time zone NOT NULL DEFAULT current_timestamp,
 	PRIMARY KEY ( hash )
 );
+CREATE TABLE wallets (
+	address bytea NOT NULL,
```

Comment:
AFAIK the current plan is to keep the secret/mnemonic out of the system and store only the public address here.

Question: how can the satellite operator collect the money? They have the wallet addresses, but they also need the private key.

Private key can be re-generated from a mnemonic and the derivation path. There are multiple options here:

 1. There is no hint to the private key, the payment system will regenerate a lot of private keys from the original mnemonic and derivation path iterator and check if the generated account address matches. (Usually this is how the wallets are working: they check if the addresses are used on the chain. See the traces of this approach [here](https://github.com/miguelmota/go-ethereum-hdwallet/blob/ca8871b71a79c01875d53a75b2a8790d2575763a/hdwallet.go#L212) )
 
2. A more effective approach is saving a unique identifier here (like the derivation path, eg. `m/44'/60'/0'/0/1`). This unique identifier is required for collecting the payments (it can be cached in satellite, too). When a list of addresses with real balance is exported from here, the derivation path (required for generating the private key) can be included in the report. **But this would require one more field here**

3. A third option: when the CLI generates the private keys, we can save the `address->derivation path` mapping to a csv which can be used during the payment collection. (I assume it will be a CLI, too). If this is csv is lost it can be recovered with 1 (but it's slow)

Do we have any plan here? (When I started to write this comment I thought that we should do 2, but now I think 3 can work as well, which means this scheme is good enough). What do you think?

Reply:
Don't think it's a security risk. We use a very standard derivation path which is easy to guess, anyway. The secret part is the mnemonic, not the derivation part.

But I am fine with 3 (==the current approach). It's less convenient for the CLI, but more flexible:  you don't need to use key derivation, technically you can also upload independent, generated addresses. But in this case it's the responsibility of the client to save the private keys (or the mnemonic + derivation path) for each address.

Reply:
Sounds good!

Reply:
https://review.dev.storj.io/c/storj/storjscan/+/7116